packages/shared/lib/core/wallet/actions/buildWalletState.ts

```diff
@@ -42,6 +44,10 @@ export async function buildWalletState(
         accountOutputs = await wallet.accounts()
         implicitAccountOutputs = await wallet.implicitAccounts()
         walletOutputs = await wallet.outputs()
+        if (accountOutputs.length && !get(mainAccountId)) {
```

Comment:
why do we set the main account when we build the wallet?

Reply:
Only set if there are accounts and no previous main account. When we do login, this is where the wallet is built, so we think it's the right place to set a default main account
Maybe we could also check that the account has a feature for block issuer and if not, not set any by default. wdyt?

Reply:
alright then, yes we should always check before setting an account that the account indeed has the block issuer feature, so first we have to know if the SDK returns with `accounts` only accounts that can issue blocks or not

Reply:
The sdk returns all the accounts, we will have to do the feature block issuer filter

Reply:
lets do some utils out of it then, you can try some kind of `getBlockIssuerAccounts(wallet)` using a `hasBlockIssuerFeature(account)` or `isBlockIsserAccount(account)` that you can use here and there